### ISSUE TITLE
feature: an option to reverse the root dir/cwd behavior

### ISSUE THREAD

**Issue by linsui** [LazyVim/LazyVim#704]
### Did you check the docs?

- [X] I have read all the LazyVim docs

### Is your feature request related to a problem? Please describe.

The root dir is sometimes not expected and the cwd is always the path set by me. So I want to use the cwd by default but all keymaps for cwd use upper case.

### Describe the solution you'd like

I'd like to have an option to reverse the behavior so that I can use lower case keymaps for cwd.

### Describe alternatives you've considered

I don't really need the root dir so if I can set the root dir to cwd by default it also works for me.

**Comment by dpetka2001** [comment]
Why don't you just change the keymaps to what you like?

**Comment by linsui** [comment]
It's verbose to change them one by one. But yes it also works.

**Comment by dpetka2001** [comment]
It might be verbose, but it's the way to do it. That's why everyone tinkers with his own config if he's not satisfied with the defaults.

**Comment by linsui** [comment]
Is it possible to reuse the default config as much as possible?

**Comment by dpetka2001** [comment]
Of course it is. Depends clearly on your needs.

**Comment by folke** [comment]
If you don't want that functionality, just do something like:

```lua
require("lazyvim.util").get_root = function() return vim.loop.cwd() end
```

**Comment by linsui** [comment]
That's what I want, thanks!

**Comment by linsui** [comment]
The trick doesn't work anymore. What should I do on LazyVim 10?

**Comment by dpetka2001** [comment]
From the [docs](https://www.lazyvim.org/configuration/general#options)
> -- LazyVim root dir detection
-- Each entry can be:
-- * the name of a detector function like `lsp` or `cwd`
-- * a pattern or array of patterns like `.git` or `lua`.
-- * a function with signature `function(buf) -> string|string[]`
vim.g.root_spec = { "lsp", { ".git", "lua" }, "cwd" }

**Comment by linsui** [comment]
Great, thanks!

**Comment by tom-james-watson** [comment]
Just to be clear, the answer to how to disable the root logic seems to be to edit e.g. `.config/nvim/lua/config/options.lua` and add:

```
vim.g.root_spec = { "cwd" }
```

This functionality doesn't really work if you're working on a monorepo that for example uses yarn workspaces. If I open a file in one workspace, everything is suddenly scoped to that workspace, even though I need to develop against multiple workspaces at once.

**Comment by folke** [comment]
This functionality has been added **specifically** to easily navigate large monorepos.
Find files/grep/explorer/... are always scoped to the specific project you are in in the monorepo.
Every keymap can also be executed with uppercase to use the cwd instead.

`<leader>fF` will find all files.

**Comment by tom-james-watson** [comment]
Fair enough - I guess it just depends on how each person works / how often you're working cross-project.
